lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -66,10 +66,68 @@ class TraceProcessor {
   }
 
   /**
+   * This method sorts a group of trace events that have the same timestamp. We want to...
+   *
+   * 1. Put E events first, we finish off our existing events before we start new ones.
```

Comment:
can a B/E pair ever have the same `ts`? If so, this would misorder them.

Reply:
If so, there's no other way for us to know at this stage either because we now know the order they came in doesn't really mean anything. We'll have to handle this on consumption side like we do with the other problematic cases then.

Reply:
> If so, there's no other way for us to know at this stage either because we now know the order they came in doesn't really mean anything. We'll have to handle this on consumption side like we do with the other problematic cases then.

hmm, but in that case wouldn't just ordering them B then E fix it? The hard part would be knowing how to connect the two

Reply:
this kinda gets at the fact that there are still lots of edge cases where we have to change our consumption. this won't particularly matter with the PR I'm about to open so I have no strong feelings here.

Reply:
> can a B/E pair ever have the same `ts`? 

so I instrumented this code a bit and ran it over a good set of gnarly traces and never found any instance where this happened *if* we differentiate by event `name` *and* `pid`.

Reply:
> so I instrumented this code a bit and ran it over a good set of gnarly traces and never found any instance where this happened _if_ we differentiate by event `name` _and_ `pid`.

Actually, `lighthouse-core/test/fixtures/traces/tracingstarted-after-navstart.json` does fail at this, but unclear if it's from us editing it :)

Reply:
this conflict should be resolved by #9491 + this existing work, so I would vote to punt, it's already captured in a test in #9491 👍